web/source/osk/banner.ts

```diff
@@ -135,31 +135,55 @@ namespace com.keyman.osk {
     }
   }
 
-  export class BannerSuggestionSpec {
-    id: string;
-    languageID: string;
-    text?: string;
-    width: number;
-    pad?: string;
-    widthpc?: number; // Added during OSK construction.
-    padpc?: number; // Added during OSK construction.
-
-    public DEFAULT_SUGGESTION_WIDTH: number = 50; // pixels
-
-    constructor(id: string, languageID: string, text?: string, width?: number, pad?: string) {
-      this.id = id;
-      this.languageID = languageID;
-      this.text = text;
-      this.width = width ? width : this.DEFAULT_SUGGESTION_WIDTH;
-      this.pad = pad;
+  export class BannerSuggestion {
+    div: HTMLDivElement;
+    private display: HTMLSpanElement;
+    private fontFamily?: string;
+
+    private suggestion: Suggestion;
+
+    private index: number;
+
+    private static readonly BASE_ID = 'kmw-suggestion-';
+
+    constructor(index: number) {
+      let keyman = com.keyman.singleton;
+
+      this.index = index;
+
+      this.constructRoot();
+
+      // Provides an empty, base SPAN for text display.  We'll swap these out regularly;
+      // `Suggestion`s will have varying length and may need different styling.
+      let display = this.display = keyman.util._CreateElement('span');
+      this.div.appendChild(display);
     }
-  }
 
-  export class BannerSuggestion {
-    spec: BannerSuggestionSpec;
+    private constructRoot() {
+      let keyman = com.keyman.singleton;
+
+      // Add OSK suggestion labels
+      let div = this.div = keyman.util._CreateElement('div'), ds=div.style;
+      div.className = "kmw-suggest-option";
+      div.id = BannerSuggestion.BASE_ID + this.index;
+
+      let kbdDetails = keyman.keyboardManager.activeStub;
```

Comment:
I was trying this branch on the Android app. Remember the weirdness where activeStub was null on startup?
That's causing kbdDetails to throw

Reply:
That'd make sense... if you were trying to show the `SuggestionBanner` without a keyboard.  I'd think that scenario should only ever happen during development tests?